src/serve/javaEmulators.ts

```diff
@@ -194,17 +232,26 @@ export async function stop(targetName: JavaEmulators): Promise<void> {
   });
 }
 
+export function downloadIfNecessary(targetName: JavaEmulators) {
```

Comment:
`export async function` that returns `Promise<void>`?

Reply:
I think declaring a function `async` without using `await` inside it is considered rude.

Reply:
The return type of this function is `Promise<something> | undefined`. If it's async, you could reduce it to `Promise<any>`. For that matter, do you even care what it returns? Maybe the `if (!hasEmulator)` should just await `downloadEmulator` and then the return type is `Promise<void>`. Thoughts?

Reply:
I think you're looking at an outdated diff, the function is now:
```ts
export function downloadIfNecessary(targetName: JavaEmulators): Promise<any> {
  const emulator = EmulatorDetails[targetName];
  const hasEmulator = fs.existsSync(emulator.binaryPath);

  if (!hasEmulator) {
    return downloadEmulator(targetName);
  } else {
    return Promise.resolve();
  }
}
```

Reply:
But anyway I can move it to `async` and await the inner ca..